lib/signs_ui/config/state.ex

```diff
@@ -262,11 +262,27 @@ defmodule SignsUi.Config.State do
       |> File.read!()
       |> Jason.decode!(keys: :atoms)
 
-    sign_stops =
+    sl_sign_stops =
+      for %{id: "Silver_Line." <> _ = id} = sign <- signs_json,
+          config_list <- [sign[:configs], sign[:top_configs], sign[:bottom_configs]],
+          config_list,
+          %{sources: sources} <- config_list,
```

Comment:
**suggestion:** if i'm reading this right i think this should work?

```suggestion
          %{sources: sources} <- [sign[:configs], sign[:top_configs], sign[:bottom_configs]],
```

Reply:
The issue here is that bus signs either have `configs` or both `top_configs` and `bottom_configs`. So line 268 is to filter out the ones the sign doesn't have.

Reply:
yep the `<-` in this `for` should be able to handle that all in one go, right?

```elixir
iex(1)> configs = %{ sources: "configs" }        
%{sources: "configs"}
iex(2)> top_configs = %{ sources: "top_configs" }
%{sources: "top_configs"}
iex(3)> bottom_configs = %{ sources: "bottom_configs" }
%{sources: "bottom_configs"}

# Will only match `sign[:configs]`
iex(4)> sign = %{ configs: configs }
%{configs: %{sources: "configs "}}
iex(5)> for %{ sources: sources } <- [sign[:configs], sign[:top_configs], sign[:bottom_configs]] do
...(5)>   sources
...(5)> end
["configs"]

# Will only match `sign[:top_configs]`
iex(6)> sign = %{ top_configs: top_configs }                                                       
%{top_configs: %{sources: "top_configs"}}
iex(7)> for %{ sources: sources } <- [sign[:configs], sign[:top_configs], sign[:bottom_configs]] do
...(7)>   sources                                                                                  
...(7)> end                                                                                        
["top_configs"]

# Will only match `sign[:bottom_configs]`
iex(8)> sign = %{ bottom_configs: bottom_configs }                                                 
%{bottom_configs: %{sources: "bottom_configs"}}
iex(9)> for %{ sources: sources } <- [sign[:configs], sign[:top_configs], sign[:bottom_configs]] do
...(9)>   sources                                                                                  
...(9)> end
["bottom_configs"]
```

Reply:
ah! @cmaddox5 has pointed out that the issue with my proposed solution is that `config_list` itself needs to be iterated over and that fails if it is `nil`.

here's what i've suggested (very non-blockingly)

**using a guard in the match:**
```suggestion
          config_list when is_list(config_list) <- [sign[:configs], sign[:top_configs], sign[:bottom_configs]],
          %{sources: sources} <- config_list,
```

**keeping with the filtering predicate but making it explicit**

```suggestion
          config_list <- [sign[:configs], sign[:top_configs], sign[:bottom_configs]],
          is_list(config_list),
          %{sources: sources} <- config_list,
```